test/Microsoft.AspNet.Mvc.Core.Test/Routing/InnerAttributeRouteTest.cs

```diff
@@ -360,6 +360,121 @@ public class InnerAttributeRouteTest
             Assert.False(context.IsHandled);
         }
 
+        [Fact]
```

Comment:
Do we need more tests with things like catch-all or complex-segments?

We didn't really need to thoroughly test this feature before because it wasn't sophisticated at all. Now we want to have tests covering more complex scenarios as validation that 'feels' right.